### ISSUE TITLE
Getting started sample gives: [MobX] Cannot obtain atom from undefined

### ISSUE THREAD

**Issue by jtheisen** [mobxjs/mobx-state-tree#1588]
**_Bug report_**

On the getting-started page of the current documentation as it shows on [mobx-state-tree.js.org](https://mobx-state-tree.js.org/intro/getting-started), the [first sample with any output](https://codesandbox.io/s/kkl8kn4pq5) gives the error message in the subject line.

It also does this in my vanilla create-react-app on this first attempt to try mst.

**Comment by ptboyer** [comment]
@jtheisen What version of `mobx` are you using?

I only started seeing this error after upgrading to `"mobx": "^6.0.0"`, whereas the documentation  for MST only mentions support for mobx 5.

**Comment by jtheisen** [comment]
@ptboyer Indeed that works, thanks a lot!

The docs' samples are not locked to a major version.

**Comment by ptboyer** [comment]
@jtheisen Glad that helped! 😃  

Not sure about the "docs' samples not being locked to a major version", but the supported major versions of MobX are stated in the [Installation](https://mobx-state-tree.js.org/intro/installation) section: "However, for MobX version 4 or 5 can be used."

**Comment by makarov-roman** [comment]
any ideas what should we do to support mobx 6?

**Comment by ptboyer** [comment]
@makarov-roman Having a little look now:

- The error "Cannot obtain atom from ..." is defined [here](https://github.com/mobxjs/mobx/blob/3053cff1145182fedcd96811b9345ff187cfc89b/src/errors.ts#L53) with error code `28`.
- Error code `28` is only referenced [here](https://github.com/mobxjs/mobx/blob/3053cff1145182fedcd96811b9345ff187cfc89b/src/types/type-utils.ts#L46) as part of `getAtom`.
- `getAtom` is currently imported by `mobx-state-tree` in:
  - `/src/utils` [here](https://github.com/mobxjs/mobx-state-tree/blob/bca6b040194f04081adeef258f402c6af9e57c36/packages/mobx-state-tree/src/utils.ts#L1)
  - `/src/types/complex-types` [here](https://github.com/mobxjs/mobx-state-tree/blob/bca6b04019/packages/mobx-state-tree/src/types/complex-types/model.ts#L7).

That's what I got so far.

**Comment by ptboyer** [comment]
`getAtom` "dies" if `thing` (first param) is `undefined`, as the error message suggests.

`getAtom` is used in `invalidateComputed` function from `/src/utils.ts`, which is called [here](https://github.com/mobxjs/mobx-state-tree/blob/bca6b04019/packages/mobx-state-tree/src/core/node/object-node.ts#L234) as `invalidateComputed(this, "snapshot")`, so is `this` undefined at the time?

There is also a "FIXME" [here](https://github.com/mobxjs/mobx-state-tree/blob/bca6b04019/packages/mobx-state-tree/src/types/complex-types/model.ts#L646):

```
getSnapshot(node: this["N"], applyPostProcess = true): this["S"] {
  ...
  this.forAllProps((name, type) => {
      // TODO: FIXME, make sure the observable ref is used!
      ;(getAtom(node.storedValue, name) as any).reportObserved()
      res[name] = this.getChildNode(node, name).snapshot
  })
  ...
}
```

... from which of `getSnapshot` the `node.storedValue` is set for a model (`object-node`) [here](https://github.com/mobxjs/mobx-state-tree/blob/bca6b04019/packages/mobx-state-tree/src/core/node/object-node.ts#L217) which could be undefined?

**Comment by Romanior** [comment]
There is a WIP PR to support Mobx6 https://github.com/mobxjs/mobx-state-tree/pull/1569. the sandbox demo is on mobx 6

**Comment by elliothershberg** [comment]
Hi! I hit this issue also. Is there a package.json that could be used to getting the sandboxes in the tutorial working again? When I tried specifying a version of Mobx, I ran into issues with the other dependencies.

**Comment by elliothershberg** [comment]
Reinstalling using: `yarn add mobx@5.15.7 mobx-react mobx-state-tree` worked for me to get the demo code to work 👍

**Comment by adamerose** [comment]
> Reinstalling using: `yarn add mobx@5.15.7 mobx-react mobx-state-tree` worked for me to get the demo code to work 👍

This led me to get this error https://github.com/mobxjs/mobx-react-lite/issues/319

I had to do `npm install mobx@5 mobx-react@6 mobx-state-tree`

The `mobx-state-tree` docs should be updated to reflect this since the latest versions of `mobx ` (6) and `mobx-react` (7) are not supported and broke all the demos.

**Comment by tejaspda** [comment]
I am still facing this issue any suggestion how to fix it from PR does it not available if I do npm install mobx-state-tree

**Comment by stfenjobs** [comment]
same here

**Comment by stfenjobs** [comment]
same error

**Comment by kuasha420** [comment]
@stfenjobs An RC is available for latest mobx/mobx-react(-lite). Try it with 

`yarn add mobx-state-tree@rc`

**Comment by dpyeates** [comment]
mobx-state-tree@rc works for me, thanks.

**Comment by eunsukimme** [comment]
`mobx(v6.0.1), mobx-react(v7.0.0), mobx-state-tree(v4.0.1-rc.1)`
In codesandbox, above versions works for me.

**Comment by justin-hackin** [comment]
Thanks for your efforts to get Mobx6 supported!  Wanted to get switched over to react-mobx-lite too so I tried installing the RC version but I encountered the following error (probably because I'm using hooks and provider to get state into my components).  Not sure which package is responsible though...

```
Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.
    at resolveDispatcher (:9080/home/cosmo/code/spacecraft-net-factory/node_modules/react/cjs/react.development.js:1465)
    at Object.useState (:9080/home/cosmo/code/spacecraft-net-factory/node_modules/react/cjs/react.development.js:1496)
    at Object.useForceUpdate (:9080/home/cosmo/code/spacecraft-net-factory/node_modules/mobx-react-lite/lib/utils/utils.js:22)
    at Object.useObserver (:9080/home/cosmo/code/spacecraft-net-factory/node_modules/mobx-react-lite/lib/useObserver.js:20)
    at :9080/home/cosmo/code/spacecraft-net-factory/node_modules/mobx-react-lite/lib/observer.js:26
    at renderWithHooks (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:14804)
    at updateFunctionComponent (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:17035)
    at updateSimpleMemoComponent (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:16973)
    at updateMemoComponent (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:16862)
    at beginWork (webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:18683)
```

**Comment by kasim444** [comment]
If you update the Mobx-state-tree package, the problem will probably be resolved. At least it was the solution for me.

`yarn upgrade mobx-state-tree --latest`

**Comment by ashokkumar88** [comment]
> Reinstalling using: `yarn add mobx@5.15.7 mobx-react mobx-state-tree` worked for me to get the demo code to work

@elliothershberg Can you please inform which version are you using for mobx-react and mobx-state.tree? I am getting error mobx.makeObservable in not a function.

**Comment by soadzoor** [comment]
I ran into this problem when upgraded from v5 to v6. The solution was to add the `makeObservable(this)` to the constructors for classes, that used the `@observer` decorator. For that I used this script: https://www.npmjs.com/package/mobx-undecorate
```npx mobx-undecorate --keepDecorators```

Unfortunately it messed up my indentations (from tabs to spaces) in the constructors, and it complained about casting in typescript (eg.: `<T> item` needed to be converted to `item as T`), but other than that, it worked fine.
